test/span_test.dart

```diff
@@ -8,15 +8,15 @@ import 'package:source_span/source_span.dart';
 import 'package:test/test.dart';
 import 'package:yaml/yaml.dart';
 
-void _expectSpan(SourceSpan source, String expected) {
-  final result = source.message('message');
+void _expectSpan(SourceSpan? source, String expected) {
+  final result = source!.message('message');
```

Comment:
It looks like this wasn't expecting the `source` to be nullable, so we should examine the call sites to understand why that would be. My guess (though this is a terrible interface for exploring the code) is that it's because `YamlNode.span` was incorrectly made nullable.

Reply:
Fixed.